File: exercicios/java/curso_programacao/src/DiagonalNegativos.java
import java.util.Scanner;

public class DiagonalNegativos {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int i, j, N, totalNegativos;
		
		System.out.print("Qual a ordem da matriz? ");
		N = sc.nextInt();
		
		int[][] mat = new int[N][N];
		
		for(i = 0; i < N; i++) {
			for(j = 0; j < N; j++) {
				System.out.print("Elemento [" + i + "," + j + "]: ");
				mat[i][j] = sc.nextInt();
			}
		}
		
		System.out.println("DIAGONAL PRINCIPAL: ");
		for(i = 0; i < N; i++) {
			System.out.print(mat[i][i] + " ");
		}
		
		System.out.println();
		
		totalNegativos = 0;
		for(i = 0; i < N; i++) {
			for(j = 0; j < N; j++) {
				if(mat[i][j] < 0) {
					totalNegativos++;
				}
			}
		}
		System.out.println("QUANTIDADE DE NEGATIVOS = " + totalNegativos);
		
		sc.close();
	}

}

File: exercicios/python/diagonal_negativos.py
N = int(input('Qual a ordem da matriz? '))

mat: [[int]] = [[0 for j in range(N)] for x in range(N)]

for i in range(0, N):
    for j in range(0, N):
        mat[i][j] = int(input(f'Elemento [{i}, {j}]: '))

print()
print('DIAGONAL PRINCIPAL: ')
for i in range(0, N):
    print(f'{mat[i][i]} ')

count: int = 0
for i in range(0, N):
    for j in range(0, N):
        if mat[i][j] < 0:
            count = count + 1

print(f'QUANTIDADE DE NEGATIVOS = {count}')
